### PR TITLE
[COLAB-2436] Fix redirect loop in Google sign in

### PR DESCRIPTION
**Pull request by jobachhu**
This PR fixes a redirect loop when google displays the account chooser (i.e. when you're signed into more than one Google account). This was caused by the AuthenticationSuccessHandler always redirecting to the referrer.

This PR also makes sure the redirect URL is always a relative url, as absolute urls don't make sense.

---
This change is [Reviewable](https://reviewable.io/reviews/cci-mit/xcolab/79)

### CODE REVIEW

File: microservices/clients/contestproposal-client/src/main/java/org/xcolab/client/proposals/ProposalAttributeClient.java [hunk 1]
```diff
@@ -52,10 +52,6 @@ public ProposalAttribute createProposalAttribute(ProposalAttribute proposalAttri
                 .execute().toPojo(serviceNamespace);
     }
 
-    public ProposalAttribute getImpactProposalAttributes(Long proposalId) {
-        return null;
-    }
-
```
Comment (umov) [inline microservices/clients/contestproposal-client/src/main/java/org/xcolab/client/proposals/ProposalAttributeClient.java]: This change seems unrelated.

File: microservices/clients/contestproposal-client/src/main/java/org/xcolab/client/proposals/ProposalAttributeClientUtil.java [hunk 2]
```diff
@@ -23,11 +23,6 @@ public static ProposalAttribute createProposalAttribute(
         return client.createProposalAttribute(proposalAttribute);
     }
 
-    public static ProposalAttribute getImpactProposalAttributes(
-            Long proposalId) {
-        return client.getImpactProposalAttributes(proposalId);
-    }
-
```
Comment (umov) [inline microservices/clients/contestproposal-client/src/main/java/org/xcolab/client/proposals/ProposalAttributeClientUtil.java]: This change also seems unrelated.

File: view/src/main/java/org/xcolab/view/pages/loginregister/LoginRegisterService.java [hunk 3]
```diff
@@ -240,7 +240,7 @@ public void logIn(HttpServletRequest request, HttpServletResponse response, Memb
         try {
             Authentication authentication =
                     authenticationService.authenticate(request, response, member);
-            authenticationSuccessHandler.onAuthenticationSuccess(request, response, authentication);
+            authenticationSuccessHandler.onAuthenticationSuccess(request, response, authentication, false);
```
Comment (umov) [inline view/src/main/java/org/xcolab/view/pages/loginregister/LoginRegisterService.java]: Where is the `false` coming from and why is it `false`? Consider moving it into a variable with an appropriate name to make it easier to understand the code.

File: view/src/test/java/org/xcolab/entity/utils/LinkUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
+package org.xcolab.entity.utils;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class LinkUtilsTest {
```
Comment (umov) [inline view/src/test/java/org/xcolab/entity/utils/LinkUtilsTest.java]: Please add a test for a blank URL string, e.g. `" "`.

File: microservices/clients/contestproposal-client/src/main/java/org/xcolab/client/proposals/ProposalAttributeClient.java (hunk 1 above)
Comment (jobachhu) [reply microservices/clients/contestproposal-client/src/main/java/org/xcolab/client/proposals/ProposalAttributeClient.java]: You're right, this commit is on the wrong branch. You can ignore this, the change has no functional impact as the method was unused.

File: microservices/clients/contestproposal-client/src/main/java/org/xcolab/client/proposals/ProposalAttributeClientUtil.java (hunk 2 above)
Comment (jobachhu) [reply microservices/clients/contestproposal-client/src/main/java/org/xcolab/client/proposals/ProposalAttributeClientUtil.java]: You can ignore this, the change has no functional impact as the method was unused.

File: view/src/main/java/org/xcolab/view/pages/loginregister/LoginRegisterService.java (hunk 3 above)
Comment (jobachhu) [reply view/src/main/java/org/xcolab/view/pages/loginregister/LoginRegisterService.java]: Done.

File: view/src/test/java/org/xcolab/entity/utils/LinkUtilsTest.java (hunk 4 above)
Comment (jobachhu) [reply view/src/test/java/org/xcolab/entity/utils/LinkUtilsTest.java]: I rewrote the function to now parse and validate all input URIs, this way this test makes more sense (as otherwise we'd just be double checking the pass-through). I think this makes the method more straight-forward: it will either return a relative URI or throw an `IllegalArgumentException`.
